src/Adapter/Order/CommandHandler/UpdateProductInOrderHandler.php

```diff
@@ -146,35 +154,21 @@ public function handle(UpdateProductInOrderCommand $command)
         // Save order detail
         $res &= $orderDetail->update();
 
-        // Update weight SUM
-        $order_carrier = new OrderCarrier((int) $order->getIdOrderCarrier());
-        if (Validate::isLoadedObject($order_carrier)) {
-            $order_carrier->weight = (float) $order->getTotalWeight();
-            $res &= $order_carrier->update();
-            if ($res) {
-                $order->weight = sprintf('%.3f ' . Configuration::get('PS_WEIGHT_UNIT'), $order_carrier->weight);
-            }
-        }
+        // Update quantity and amounts
+        $order = $this->orderProductUpdater->update(
```

Comment:
Shouldn't the OrderDetail updates (performed on L140) be done in the OrderProductUpdater as well? Along with Customization updates?

Also I might be wrong but it would seem more logical to make the order details updates before updating the order no? And we could could maybe use the OrderAmountUpdater service as well here to clean a big piece of code? (total computing and all?)

Reply:
Oh actually I just realized that `OrderProductUpdater` already uses `OrderAmountUpdater` which is even better!

I also understood that the total udpates in this class are related to the invoice not the order, since it's the invoice related to the OrderDetail maybe it should be done in the service as well?

Reply:
done